### PR TITLE
fix: nullptr exception

### PR DESCRIPTION
**Pull request by tellundro**
fix for exception on line 364 due to _lifted hashmap not being initialized.

### CODE REVIEW

**Comment by kamronbatman** [comment]
@tellundro, I am thinking we should do a null check instead, and then initialize before adding. This way we aren't allocating for the HashSet until we need it.

**Comment by tellundro** [comment]
@kamronbatman yeah that makes sense. I'll edit the code and push it again.

File: Projects/UOContent/Items/Containers/TreasureMapChest.cs [hunk 1]
```diff
@@ -361,6 +361,11 @@ private bool CheckLoot(Mobile m, bool criminalAction)
 
     public override void OnItemLifted(Mobile from, Item item)
     {
+        if (_lifted == null)
```
Comment (kamronbatman) [inline Projects/UOContent/Items/Containers/TreasureMapChest.cs:364]: Let's change to this to:
```cs
_lifted ??= new HashSet<Item>();
```

And move it right above this line:
```cs
_lifted.Add(item);
```

**Review by kamronbatman — CHANGES_REQUESTED** [review]

File: Projects/UOContent/Items/Containers/TreasureMapChest.cs (hunk 1 above)
Comment (Tultra) [reply Projects/UOContent/Items/Containers/TreasureMapChest.cs:364]: @kamronbatman I guess we need to keep it above 

`var notYetLifted = !_lifted.Contains(item);`

which is where the exception happened.

File: Projects/UOContent/Items/Containers/TreasureMapChest.cs [hunk 2]
```diff
@@ -361,7 +361,10 @@ private bool CheckLoot(Mobile m, bool criminalAction)
 
     public override void OnItemLifted(Mobile from, Item item)
     {
-        var notYetLifted = !_lifted.Contains(item);
+        
+        _lifted ??= new HashSet<Item>();
+        
+        var notYetLifted = !_lifted.Contains(item);
```
Comment (kamronbatman) [inline Projects/UOContent/Items/Containers/TreasureMapChest.cs:367]: Let's add a null check here, and move the assignment to where we need it (above the add):

```cs
var notYetLifted = _lifted?.Contains(item) != true;
```

File: Projects/UOContent/Items/Containers/TreasureMapChest.cs [hunk 3]
```diff
@@ -361,15 +361,12 @@ private bool CheckLoot(Mobile m, bool criminalAction)
...
-        _lifted ??= new HashSet<Item>();
-        
-        var notYetLifted = !_lifted.Contains(item); 
-
+        var notYetLifted = _lifted?.Contains(item) != true; 
         from.RevealingAction();
 
         if (notYetLifted)
         {
+            _lifted = new HashSet<Item>();
```
Comment (kamronbatman) [inline Projects/UOContent/Items/Containers/TreasureMapChest.cs:369]: This needs to be `??=`

**Review by kamronbatman — APPROVED** [review]

**Pull request merged**